azurerm/internal/services/compute/virtual_machine_scale_set_resource.go

```diff
@@ -135,9 +135,9 @@ func resourceArmVirtualMachineScaleSet() *schema.Resource {
 				Type:     schema.TypeString,
 				Required: true,
 				ValidateFunc: validation.StringInSlice([]string{
-					string(compute.Automatic),
+					string(compute.AutomaticByOS),
 					string(compute.Manual),
-					string(compute.Rolling),
+					string(compute.AutomaticByPlatform),
```

Comment:
Same here, should be `compute.UpgradeModeAutomatic`, `compute,UpgradeModeManual` and `compute.UpgradeModeRolling`